File: sunbin/day22/002_인덱스바꾸기.js
//https://school.programmers.co.kr/learn/courses/30/lessons/120895

function solution(my_string, num1, num2) {

  const arr = my_string.split('');

  a = arr[num1]
  b = arr[num2]

  arr.splice(num1, 1, b)
  arr.splice(num2, 1, a)

  const answer = arr.join('')

  return answer;
}

// 다른 풀이: 구조 분해 할당
function solution(my_string, num1, num2) {
  my_string = my_string.split('');
  [my_string[num1], my_string[num2]] = [my_string[num2], my_string[num1]];
  return my_string.join('');
}

File: sunbin/Lv0/day22/python/002_인덱스바꾸기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/120895

def solution(my_string, num1, num2):
    str_list = list(my_string)

    a = str_list[num1]
    b = str_list[num2]

    str_list[num1] = b
    str_list[num2] = a

    answer = "".join(str_list)

    return answer


# 다른 풀이: 튜플
def solution_b(my_string, num1, num2):
    s = list(my_string)
    s[num1], s[num2] = s[num2], s[num1]

    return ''.join(s)


solution("hello", 1, 2)
